File: source/MyExpenses/MyExpenses.Web/server/model/Expenses/Employee.js
'use strict';

let fullName = function () {
    return this.firstName + ' ' + this.lastName;
};

module.exports = function(sequelize, DataTypes) {
    return sequelize.define('Employee', {
        id: {
            type: DataTypes.INTEGER,
            allowNull: false,
            primaryKey: true,
            autoIncrement: true 
        },
        firstName: {
            type: DataTypes.STRING,
            allowNull: false
        },
        lastName: {
            type: DataTypes.STRING,
            allowNull: false
        },
        email: {
            type: DataTypes.STRING,
            allowNull: false
        },
        identifier: {
            type: DataTypes.STRING,
            allowNull: false
        },
        jobTitle: {
            type: DataTypes.STRING,
            allowNull: false
        },        
        teamId: {
            type: DataTypes.INTEGER,
            allowNull: false,            
            references: {
                model: 'Team',
                key: 'Id'
            }            
        },
        isTeamManager: {
            type: 'BIT',
            allowNull: false
        },
        bankAccountNumber: {
            type: DataTypes.STRING,
            allowNull: false,

        }
    }, {
        timestamps: false,
        tableName: 'Employee',
        schema: 'Expense',
        instanceMethods: {
            fullName: fullName
        }
    });
};

File: source/MyExpenses/MyExpensesDataGenerator/DataModel/Employee.cs
namespace MyExpensesDataGenerator
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Expense.Employee")]
    public partial class Employee
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Employee()
        {
            Bonifications = new HashSet<Bonification>();
            Bonifications1 = new HashSet<Bonification>();
            ExpenseReports = new HashSet<ExpenseReport>();
            Pictures = new HashSet<Picture>();
        }

        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(50)]
        public string LastName { get; set; }

        [Required]
        [StringLength(60)]
        public string Email { get; set; }

        [Required]
        [StringLength(50)]
        public string Identifier { get; set; }

        [Required]
        [StringLength(50)]
        public string JobTitle { get; set; }

        public int TeamId { get; set; }

        public bool IsTeamManager { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Bonification> Bonifications { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Bonification> Bonifications1 { get; set; }

        public virtual Team Team { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<ExpenseReport> ExpenseReports { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Picture> Pictures { get; set; }

        public virtual EmployeePurchase EmployeePurchase { get; set; }

        [Required]
        [StringLength(17)]
        public string BankAccountNumber { get; set; }
    }
}
